redbeams/src/main/java/com/sequenceiq/redbeams/service/network/AwsSubnetChooser.java

```diff
@@ -0,0 +1,88 @@
+package com.sequenceiq.redbeams.service.network;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimaps;
+import com.sequenceiq.cloudbreak.cloud.model.CloudSubnet;
+
+@Component
+public class AwsSubnetChooser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsSubnetChooser.class);
+
+    private static final int MIN_SUBNET_IN_DIFFERENT_AZ = 2;
+
+    private static final int MAX_SUBNET_IN_DEFFERENT_AZ = 3;
+
+    @Inject
+    private AwsSubnetValidator awsSubnetValidator;
+
+    public List<CloudSubnet> chooseSubnets(List<CloudSubnet> subnetMetas) {
+        LOGGER.debug("Subnets to choose from: [{}]", subnetMetas);
+        awsSubnetValidator.validate(subnetMetas, MIN_SUBNET_IN_DIFFERENT_AZ);
```

Comment:
this should be depend on the deployment config

Reply:
we won't address it in this pr